structure/bitmap.go

```diff
@@ -0,0 +1,231 @@
+package structure
+
+import (
+	"bytes"
+	"errors"
+	"github.com/ByteStorage/FlyDB/config"
+	"github.com/ByteStorage/FlyDB/engine"
+	_const "github.com/ByteStorage/FlyDB/lib/const"
+	"github.com/bits-and-blooms/bitset"
+)
+
+type BitmapStructure struct {
+	db *engine.DB
+}
+type BitOperation string
+
+const (
+	BitAndOperation BitOperation = "AND"
+	BitOrOperation  BitOperation = "OR"
+	BitXorOperation BitOperation = "XOR"
+	BitNotOperation BitOperation = "NOT"
+)
+
+func NewBitmap(options config.Options) (*BitmapStructure, error) {
+	db, err := engine.NewDB(options)
+	if err != nil {
+		return nil, err
+	}
+	return &BitmapStructure{db: db}, nil
+}
+
+func (b *BitmapStructure) SetBit(key []byte, off uint) error {
```

Comment:
The new data structure is to make it easier for users to call the db api without having to build the corresponding data structure themselves.

Therefore, for ease of use, you should not use []byte as the input parameter of the key, but should use string. Similarly, get should not directly return []byte as much as possible. I carefully considered the type of return value when developing string and hash and finally selected interface as the corresponding return type. What's your understanding? We can discuss it.

Reply:
In this implementation, the `GetBits` function only returns data of type `[]byte,` and we have three options to choose from:
- We can continue to return a byte array of our bitset, as we currently do.
- We can return an instance of the `bitset` library, although this may not be the best choice.
- We can return absolute values in either `[]uint` or `string` format.
Each option has advantages and disadvantages, which we must carefully evaluate.

Reply:
For some personal reasons this morning, I had another core developer on our team review your code, so let's discuss it in detail.

Here are some of my thoughts：

1. First of all, I would not choose to use the second option in my mind, introducing a `bitset` library as a dependency would increase the complexity of the project and could lead to compatibility issues with other libraries or systems. The `bitset` library may provide some functionality beyond the actual needs of the project, resulting in unnecessary performance and resource overhead.
2. As for your first point, I think we propose a new data structure to make it easier for users to use a more intuitive data representation to invoke the underlying api of our database, without having to convert the data form back and forth. In addition, complex `bitset` operations on byte arrays may require more complex logic to be redefined, which is inconvenient for subsequent maintenance.
3. Personally, I prefer the third option. Use the `[]uint` or `string` format to return absolute values for greater versatility and flexibility. Returning different data types makes it easier to integrate with other libraries or systems and interact with other data formats. This flexibility makes it easier to meet user needs.
4. However, there are some problems to consider, such as memory overhead, and using absolute values may require more storage space. Especially for large bit sets, converting a `bitset` to an absolute value can increase memory consumption. This can be a problem in resource-constrained environments.

How do you see it, or which option do you prefer?

Reply:
@qishenonly Having a wrapper around the library would be something like this:
```go
type BitSet struct {
	b *bitset.BitSet
}
func newBitSet(d []byte) (*BitSet, error) {
	b := bitset.New(1)
	err := b.UnmarshalBinary(d)
	if err != nil {
		return nil, err
	}
	return &BitSet{b: b}, nil
}

func (b *BitSet) At(pos uint) bool {
	return b.b.Test(pos)
} 
```

Which, as you can see, can implement some of the necessary functions to try to traverse bitsets easily.

As you mentioned, the other option has the problem of consuming more memory, up to 64x the memory usage of just using the bitset. Either way, we have to agree with one of the solutions.

Reply:
The wrapper implementation you provide is a good way to work with `bitset`. It encapsulates the underlying `bitset` library and provides additional methods to make it easier to traverse and access `bitsets`. By creating custom `BitSet` structs and defining methods such as `A`t and `Next`, you can abstract functionality and provide a more intuitive interface for bitset manipulation.

If you can use this off-the-shelf library and still have efficient performance without affecting the integration of the environment, I think it is a good choice.

The project's storage engine is based on `bitcask` papers, which are built to maintain indexes in the database in memory, and using absolute value returns can lead to higher overhead costs. But now that memory capacity is not a big issue, I recommend using your own encapsulated bitmap method rather than using other libraries to increase the dependency cost when performance is similar.

Reply:
All right, It might cause you some trouble.